internal/transport/transport_test.go

```diff
@@ -2753,24 +2754,25 @@ func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
 // signaled or a timeout occurs.
 type hangingConn struct {
 	net.Conn
-	hangConn     chan struct{}
-	startHanging *atomic.Bool
+	hangConn chan struct{}
 }
 
 func (hc *hangingConn) Write(b []byte) (n int, err error) {
 	n, err = hc.Conn.Write(b)
-	if hc.startHanging.Load() == true {
-		// Hang the Write for more than goAwayLoopyWriterTimeout
-		timer := time.NewTimer(time.Millisecond * 5)
-		defer timer.Stop()
-		select {
-		case <-hc.hangConn:
-		case <-timer.C:
-		}
+	if n == goAwayFrameSize { // hang the conn after the goAway is received
```

Comment:
Why did you remove the atomic?

This is worse, since you're just assuming that: 1. Nothing else written will be the size of the GOAWAY frame, and 2. the GOAWAY will be written to the connection, by itself, in a single operation.